### DISCUSSION TITLE
Is there a way to change CTk.Combobox's background color when it's on 'readonly' state?

### DISCUSSION THREAD

**Discussion by duruburak** [TomSchimansky/CustomTkinter#816] [Q&A] [answered]
First image is when it's on 'normal' state. Background is gray (dark mode) by default and it's also changeable by 'fg_color' attribute.

But second image is when it's on 'readonly' state. Background suddenly changes to white and it's not changeable by 'fg_color'. It might be a bug tho, is there any way to fix this?

![31](https://user-images.githubusercontent.com/100048974/206813413-64ef41b7-b6ef-415d-acf8-e4211a7d4b91.png)
![32](https://user-images.githubusercontent.com/100048974/206813418-5dc0cfe0-b656-4e0e-8118-2ec5673b287f.png)

**Comment by Beinish** [comment]
I was about to post the same question. I actually didn't know that `state='readonly'` was causing this, so thanks :)

Hoping someone has a fix.

**Reply by duruburak** [reply]
I'm quite sure it's a bug. Could you please enlighten us? @TomSchimansky

**Comment by Tornado300** [comment]
I have the same problem

**Comment by Tornado300** [comment]
has someone solved the problem ?

**Reply by Akascape** [reply]
Just add `widget._entry.configure(readonlybackground=widget._apply_appearance_mode(widget._fg_color))`, where **widget** is your combobox.

**Comment by Akascape** [comment, marked as answer]
@duruburak, @Tornado300  These types of bugs are easy to solve 🥱,
```python
optionmenu = customtkinter.CTkComboBox(root, state="readonly", values=["A", "B", "C", "D"])
optionmenu._entry.configure(readonlybackground=optionmenu._apply_appearance_mode(optionmenu._fg_color))
optionmenu.pack()
```
@TomSchimansky , this bug can be solved by adding the `readonlybackground` parameter in the entry of combobox.
